### PR TITLE
cf integration git auth

### PR DESCRIPTION
**Pull request by andrii-codefresh**

### CODE REVIEW

File: cmd/commands/integrations.go [hunk 1]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunGitAuthCommand(cmd.Context(), cmd)
+		},
+	}
+
+	return cmd
+}
+
+func RunGitAuthCommand(ctx context.Context, cmd *cobra.Command) error {
+	user, err1 := cfConfig.GetCurrentContext().GetUser(ctx)
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: rename from err1

File: cmd/commands/integrations.go [hunk 2]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+	if err1 != nil {
+		return err1
+	}
+
+	userId := user.ID
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: dont think that you need this variable

File: cmd/commands/integrations.go [hunk 3]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+	var accountId string
+	for i := range user.Accounts {
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: move it to account util, or another util

File: cmd/commands/integrations.go [hunk 4]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+		if user.Accounts[i].Name == user.ActiveAccountName {
+			accountId = user.Accounts[i].ID
+			break
+		}
+	}
+	runtimeName := cmd.Flag("runtime").Value.String()
+	runtime, err2 := cfConfig.NewClient().V2().Runtime().Get(ctx, runtimeName)
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: err2 -> err

File: cmd/commands/integrations.go [hunk 5]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+	if err2 != nil {
+		return err2
+	}
+
+	var b bytes.Buffer
+	if !strings.HasPrefix(*runtime.IngressHost, "http") {
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: move it to util, like OPenBrowser function

File: cmd/commands/integrations.go [hunk 6]
```diff
@@ -405,6 +409,57 @@ func getAppProxyClient(runtime *string, client *sdk.AppProxyAPI) func(*cobra.Com
...
+		b.WriteString("http://")
+	}
+	b.WriteString(*runtime.IngressHost)
+	b.WriteString("/app-proxy/api/git-auth/github?user=" + userId + "&account=" + accountId)
+
+	url := b.String()
+	err3 := browser.OpenURL(url)
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go]: err3 -> err

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: cmd/commands/integrations.go [hunk 7]
```diff
@@ -29,6 +26,8 @@ import (
 	"github.com/juju/ansiterm"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"os"
```
Comment (pasha-codefresh) [inline cmd/commands/integrations.go:29]: this will cause linter issues

**Review by pasha-codefresh — APPROVED** [review]

File: pkg/util/util.go [hunk 8]
```diff
@@ -218,6 +221,35 @@ func KubeCurrentServer() (string, error) {
...
+	return "", fmt.Errorf("account id for \"%s\" not found", user.ActiveAccountName)
+}
+
+func OpenBrowserForGitLogin(ingressHost string, user string, account string) error {
+	var b bytes.Buffer
+	if !strings.HasPrefix(ingressHost, "http") {
+		b.WriteString("https://")
+	}
+	b.WriteString(ingressHost)
+	b.WriteString("/app-proxy/api/git-auth/github?userId=" + user + "&accountId=" + account)
```
Comment (roi-codefresh) [inline pkg/util/util.go:240]: You should use `https://pkg.go.dev/net/url`

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**